pkg/cmd/step/create/step_create_task.go

```diff
@@ -1293,7 +1299,11 @@ func (o *StepCreateTaskOptions) setBuildVersion(pipelineConfig *jenkinsfile.Pipe
 		if err != nil {
 			return err
 		}
-		o.Revision = "v" + version
+		if o.UseBranchAsRevision {
+			o.Revision = o.Branch
+		} else {
+			o.Revision = "v" + version
+		}
 	} else {
 		// lets use the branch name if we can find it for the version number
 		branch := o.Branch
```

Comment:
So here we use the branch already, right? Which is the pr pipelinetype I guess. So changing the pipeline type is not an option?

Reply:
Yeah, the use case still wants to have "normal" `pullrequest` pipeline builds, so we're a little stuck. I considered just creating a new pipeline type, but that's a pretty complex thing. We can revisit later, though.

Reply:
> Yeah, the use case still wants to have "normal" pullrequest pipeline builds, so we're a little stuck. I considered just creating a new pipeline type, but that's a pretty complex thing. We can revisit later, though.

that is my thinking as well. Is that really so complex. My fear is that we keep digging us deeper and deeper in.

Reply:
So my thinking is that if there are any more tweaks like this needed in the future, we say "yeah, this needs a new pipeline type" - but I feel like just this one flag is small enough to not merit that work yet.

Reply:
Where is the complexity in creating a new type? Because of the jenskins-x.yaml syntax and the fact that the release kind is "hard coded"?

What I mean is:
```yaml
  pipelines:
    release:
      pipeline:
```

Here we have a fixed "release" pipeline, which would be opposed to something more flexible like:

```yaml
  pipelines:
      pipeline:
     kind: "release"
```

Is this the issue? Could we not have hard coded "release" and "pullrequest", but then have "free form" pipelines of the type above? 

From a metapipline point of view a new pipeline type does not seem to be hard, right?

Reply:
It means going through everywhere that we refer to or use pipeline types and adding the new one, adding support for it in metapipeline/`step_create_task`, and most importantly, deciding what exactly will distinguish the new pipeline type from the others. If we're going to add a new pipeline type, I want to know exactly _why_ we're adding it, how it differs from the others, what the use case is for it, etc. i.e., I don't want to create another `feature`, given that literally no one knows what you would use _that_ for. =)

Reply:
...which is why I feel that adding these two flags to metapipeline, which can only be used by something that's calling metapipeline programmatically, is less intrusive/complex/etc. Basically, I'm fine with a small addition of complexity for "internal" APIs for a specific use case, but less so for anything that involves changing the _userfacing_ APIs - i.e., what you can do in a `jenkins-x.yml` and what that means behaviorally.